### PR TITLE
Fix #206: Remove color-darkening computation.

### LINKED ISSUE #206: Create color & icon profiles for thumbnails
We currently define thumbnails based on a single color and an entry in a drawable enum, however there are other aspects necessary to define for rendering thumbnails:
- Drawable padding/size (centerCrop works well for rendering these, but extra whitespace causes them to look a bit inconsistent)
- Background/foreground panel colors (these should be set such that they look correct with certain drawables, have proper contrast, and fit our theme)
- Text color if certain text colors don't work with certain background/foreground panel colors
- Dark mode adjustments

These properties need to be defined in a way that has some consistency with Oppia web, but without the backend needing to understand all of these properties.

### PR DESCRIPTION
**Pull request by viktoriias**
## Explanation

Fixes #206: Remove TODO comments, remove dead code. For the background color, I used `@color/topicTitleBackground` (same as in other layouts).

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by viktoriias** [comment]
Before:


After:


[Mocks](https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/92becc49-3a10-498b-885c-4cde95f73d51/)

**Comment by viktoriias** [comment]
@BenHenning Please review.

File: app/src/main/java/org/oppia/android/app/home/topiclist/TopicSummaryViewModel.kt [hunk 1]
```diff
@@ -29,8 +21,6 @@ class TopicSummaryViewModel(
   @ColorInt
```
Comment (viktoriias) [inline app/src/main/java/org/oppia/android/app/home/topiclist/TopicSummaryViewModel.kt]: I think I can remove background color from here too.

**Comment by BenHenning** [comment]
@rt4914 can you confirm whether we should be using the standard topic tile colors for the walkthrough experience?

File: app/src/main/java/org/oppia/android/app/home/topiclist/TopicSummaryViewModel.kt (hunk 1 above)
Comment (BenHenning) [reply app/src/main/java/org/oppia/android/app/home/topiclist/TopicSummaryViewModel.kt]: I think you're right--I removed this locally in an in-progress PR. Please go ahead and remove it here. Seems like a nice opportunistic fix.

File: app/src/main/res/layout/walkthrough_topic_summary_view.xml [hunk 2]
```diff
@@ -22,8 +22,8 @@
 
     <org.oppia.android.app.customview.LessonThumbnailImageView
       android:id="@+id/walkthrough_topic_thumbnail_image_view"
-      android:layout_width="match_parent"
-      android:layout_height="92dp"
+      android:layout_width="0dp"
+      android:layout_height="0dp"
```
Comment (BenHenning) [inline app/src/main/res/layout/walkthrough_topic_summary_view.xml:26]: Why this change?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @viktoriias! Left some comments. I think this is overall a nice cleanup, but we need confirmation from @rt4914 about the coloring change before submitting.

Also for note, I'm adding a background color check for the walkthrough in #2983 which will help guard against regressions from the behavior change introduced here.

File: app/src/main/res/layout/walkthrough_topic_summary_view.xml (hunk 2 above)
Comment (viktoriias) [reply app/src/main/res/layout/walkthrough_topic_summary_view.xml:26]: I noticed that "First Test Topic"'s image looked different, and these were layout dimensions in another view I used to compare, I think it was "topic_summary_view.xml"

**Comment by rt4914** [comment]
> @rt4914 can you confirm whether we should be using the standard topic tile colors for the walkthrough experience?

@BenHenning As per the [mocks](https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/92becc49-3a10-498b-885c-4cde95f73d51/) we do want to use standard colors.

File: app/src/main/res/layout/walkthrough_topic_summary_view.xml (hunk 2 above)
Comment (rt4914) [reply app/src/main/res/layout/walkthrough_topic_summary_view.xml:26]: Along with this change please use `android:lines="1"` and `android:ellipsize="end"` in `walkthrough_lesson_count_text_view` textview then these solutions will work correctly together.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@viktoriias 
I have suggested one change else everything is good.

File: app/src/main/res/layout/walkthrough_topic_summary_view.xml (hunk 2 above)
Comment (viktoriias) [reply app/src/main/res/layout/walkthrough_topic_summary_view.xml:26]: Done.

**Review by rt4914 — APPROVED** [review]
Thanks. LGTM

File: app/src/main/java/org/oppia/android/app/home/topiclist/TopicSummaryViewModel.kt (hunk 1 above)
Comment (viktoriias) [reply app/src/main/java/org/oppia/android/app/home/topiclist/TopicSummaryViewModel.kt]: Done.

**Review by BenHenning — APPROVED** [review]
Thanks @viktoriias! I like seeing unnecessary code get deleted. :)
